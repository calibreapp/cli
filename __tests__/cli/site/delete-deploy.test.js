import {
  runCLI,
  setupIntegrationServer,
  teardownIntegrationServer
} from '../../utils'

import deleteDeploy from '../../fixtures/deleteDeploy.json'

describe('site delete-deploy', () => {
  beforeAll(async () => await setupIntegrationServer(deleteDeploy))
  afterAll(async () => await teardownIntegrationServer())

  test('requires uuid', async () => {
    const out = await runCLI({
      args: 'site delete-deploy --site=test',
      testForError: true
    })
    expect(out).toMatchSnapshot()
  })

  test('deletes deploy', async () => {
    const out = await runCLI({
      args: 'site delete-deploy --site=test --uuid=1',
      testForError: true
    })
    expect(out).toMatchSnapshot()
  })
})
