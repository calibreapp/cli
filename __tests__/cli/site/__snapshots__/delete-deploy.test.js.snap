// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`site delete-deploy deletes deploy 1`] = `
"- Connecting to Calibre
âœ” Deploy deleted: 8c68b9f6-b5bf-49bc-abd7-896db6338818"
`;

exports[`site delete-deploy requires uuid 1`] = `
"

 Missing required argument: uuid 

 --- Command documentation
 calibre site delete-deploy [options]

Deletes a deploy from a site

Options:

  --help     Show help                                                 [boolean]
  --version  Show version number                                       [boolean]
  --uuid     The uuid of the deploy                                   [required]
  --site     The identifying slug of a site                  [string] [required]
  --confirm  Confirm the deletion
  --json     Return the output in JSON format"
`;
